native/exe/ServiceControlLaunch/ServiceControlLaunch.cpp

```diff
@@ -126,44 +125,71 @@ int APIENTRY WinMain(HINSTANCE hInstance,
 	DWORD dwRead = 0;

```

Comment:
Can you get rid of this line since dwRead is never actually used? And then get rid of that bad 'hsize' variable and inline the sizeof() calls whenever hsize is used...that'll be so much cleaner.

Reply:
The RegQueryValueEx where hsize is used all require LPDWORD.
Is it best to cast them or just tidy up the hsize definition?

Reply:
Whoops, my bad. I forgot that's why we had to use a temp variable like this. Do NOT just cast it as that'd be totally wrong and almost definitely crash things. It's expecting a pointer to the value because it wants to know the size on input, and then I *think* it also sets that value to the returned size on output (which is why it needs a pointer).  So just put it back the way it was.

Reply:
Didn't think it would be right trying to cast it, so I had just done a tidy up. :)

Reply:
I don't see any purpose for dwRead.  I think those lines could just be changed to "DWORD hsize = sizeof(currVer);".  The LPDWORD type is fine since that's what "&hsize" is and as Jeff pointed out, that function wants a pointer to a DWORD.  It holds the size of the buffer (currVer) provided to the function and the function returns the actual byte count put into the buffer in hsize.  Refer to the lpcbData parameter here: https://docs.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regqueryvalueexa

Reply:
Thanks, that's what I've done, just didn't want to do another commit until I'd knew what I should be doing with sprintf_s @106
Trying to keep the commits down below 5 for this change :)